packages/functional-tests/src/tests/video-test.ts

```diff
@@ -0,0 +1,103 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import * as MRE from '@microsoft/mixed-reality-extension-sdk';
+import { App } from '../app';
+import { Test } from '../test';
+
+export default class VideoTest extends Test {
+    public expectedResultDescription = "Play a couple youtube videos. Click to cycle.";
+
+    constructor(app: App, baseUrl: string, user: MRE.User) {
+        super(app, baseUrl, user);
+    }
+
+    private _state = 0;
+
+    public async run(root: MRE.Actor): Promise<boolean> {
+        const video = await MRE.Actor.CreateEmpty(this.app.context, {
+            actor: {
+                parentId: root.id,
+                name: 'video',
+                transform: {
+                    local: {
+                        position: { x: 0, y: 1, z: 0 },
+                        scale: { x: 2, y: 2, z: 2 }
+                    }
+                },
+            }
+        });
+
+        const videoStreamPromise1 = this.app.context.assetManager.createVideoStream(
+            'group1',
+            { videoSourceType: MRE.VideoSourceType.YouTube,
+                 uri: `https://www.youtube.com/watch?v=z1YNh9BQVRg` }
+        );
+        const videoStreamPromise2 = this.app.context.assetManager.createVideoStream(
+            'group1',
+            { videoSourceType: MRE.VideoSourceType.YouTube,
+                 uri: `https://www.youtube.com/watch?v=UowkIRSDHfs` }
+        );
+        let videoInstance: MRE.ForwardPromise<MRE.SoundInstance>;
+        const cycleState = () => {
+            if (this._state === 0) {
+                this.app.setOverrideText("Playing Movie 1!");
+                videoInstance = video.startSound(videoStreamPromise1.value.id,
```

Comment:
>startSound [](start = 38, length = 10)

This is really confusing.

Reply:
I think an API like this would work better:
```ts
    const video = this.context.CreateVideoPlayer(...); // Same parameters as CreateEmpty
    video.play(videoStream.id, { ...optionalArgs });
    // etc.
```

Reply:
Oh, I even read the description and still tripped on this one :)

Reply:
It was super confusing that I named the actor "video". I renamed that. The example will then become parentActor.startVideoStream() instead of video.startSound().